src/fatx_internal.h

```diff
@@ -88,10 +82,12 @@
  */
 #pragma pack(1)
 struct fatx_superblock {
+    uint32_t signature;
     uint32_t volume_id;
     uint32_t sectors_per_cluster;
-    uint16_t root_cluster;
-    uint32_t unknown1;
+    uint32_t root_cluster;
+    uint16_t unknown1;
+    uint8_t  padding[4078];
```

Comment:
`root_cluster` is supposed to be a 32bit field in the superblock.

The `unknown1` is technically supposed to be a unicode volume name of some length but I've only ever seen it as a single null terminator `\x00\x00` in retail.

Maybe it is used on memory cards? But I don't think it really matters. either way.